src/wh_nvm.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * src/wh_flash.c
+ *
+ * Simple helper library
+ *
+ */
+
+#include <stddef.h>     /* For NULL */
+#include <string.h>     /* For memset, memcpy */
+
+#include "wolfhsm/wh_common.h"
+#include "wolfhsm/wh_error.h"
+
+#include "wolfhsm/wh_nvm.h"
+
+
+int wh_Nvm_Init(whNvmContext* context, const whNvmConfig *config)
+{
+    if (    (context == NULL) ||
+            (context->cb == NULL) ) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* No callback? Return ABORTED */
+    if (context->cb->Init == NULL) {
+        return WH_ERROR_ABORTED;
+    }
+    return context->cb->Init(context->context, config);
+}
```

Comment:
When this is invoked, the context that is passed in is actually the one from the config, rather than the one that is in the upper level object.  We invoke init on this using the config to prove it's going to work.  So the binding that you are talking about has to happen at the next level up because we are not given the "real" context here, only the one from the config.  Maybe I should crank out a picture to better document this because if we are getting confused, then our customers will be too.

Reply:
Yes please, some documentation on this would be great. It is the paradigm we are using for a lot of the layers so would not be wasted effort to diagram it